server/src/main/java/org/elasticsearch/cluster/metadata/MetadataMigrateToDataStreamService.java

```diff
@@ -210,12 +223,17 @@ static void prepareBackingIndex(
             imb.removeAlias(dataStreamName);
         }
 
-        b.put(
-            imb.settings(Settings.builder().put(im.getSettings()).put("index.hidden", "true").build())
-                .settingsVersion(im.getSettingsVersion() + 1)
-                .mappingVersion(im.getMappingVersion() + 1)
-                .putMapping(new MappingMetadata(mapper))
-        );
+        Settings.Builder settingsUpdate = Settings.builder().put(im.getSettings()).put("index.hidden", "true");
```

Comment:
Nit: I'm personally more a fan of using constants (`IndexMetadata.SETTING_INDEX_HIDDEN`) and using the boolean value itself (`true` instead of `"true"`).

Reply:
I definitely agree with that - Originally I had this set in the failure store definition class but removed it. Seems I didn't tag back up on this one.